smarts/core/utils/file.py

```diff
@@ -109,6 +113,25 @@ def file_md5_hash(file_path: str) -> str:
     return str(hasher.hexdigest())
 
 
+def pickle_hash(obj, include_version=False) -> str:
+    """Converts a Python object to a hash value. NOTE: NOT stable across different Python versions."""
+    pickle_bytes = pickle.dumps(obj, protocol=4)
```

Comment:
There is no guarantee that the pickle bytecode will be the same, only that it will result in the same object after unpickling. See here: https://death.andgravity.com/stable-hashing#pickle-not-stable